### ISSUE TITLE
Document how to add dialect keywords get_columns() column information

### ISSUE THREAD

**Issue by Kawaamai** [sqlalchemy/sqlalchemy#5139]
Currently sqlalchemy supports `dialect.get_columns()` returning dialect keywords via the `dialect_options` key in the form `<dialect_name>_<argument_name>`. See the [mssql dialect implementation](https://github.com/sqlalchemy/sqlalchemy/blob/47202abbf9823e1058e0b88ce64ffd3b88027e96/lib/sqlalchemy/dialects/mssql/base.py#L2726) as an example. It would be nice to document this keyword in `dialect.get_columns()`.

I also wonder the handling of dialect keywords should be aligned with how `DialectKwArgs` works, either by changing `dialect_options` to be nested and keyed to `<dialect_name>` and `<argument_name>`, or by renaming it to `dialect_kwargs`.

Happy to work on any of this if you agree.

**Comment by zzzeek** [comment]
likely it should be documented here:

https://github.com/sqlalchemy/sqlalchemy/blob/47202abbf9823e1058e0b88ce64ffd3b88027e96/lib/sqlalchemy/engine/reflection.py#L327 

and then everything relevant in interfaces.py should refer to the reflection.py file.   having the same docs in two places is not going to be maintainable.

I'm not following the DialectKWArgs part - that class supports both the flattened and the nested version of things at the same time.    We defintiely can change nothing in the dialect API regarding this as there are dozens of 3rd party dialects using it.

**Comment by Kawaamai** [comment]
Documenting there makes sense, same to the no to API changes, got ahead of myself there.

Will see if I can find some time to make a pr for this.

**Comment by Gambit-Checkmate** [comment]
Hi, I’d like to improve the dialect documentation for get_columns() keyword information. I can add a concise example based on the current API and verify it against the existing docs build. Is this still open for an external PR?

**Comment by zzzeek** [comment]
yes the open for pull requests label allows the PR to come through

**Comment by LeonxLJX** [comment]
I'd like to take this one.

Reading the issue, this looks like a documentation-clarity gap rather than a code defect: the current page doesn't give readers enough to self-diagnose, which is why it surfaces here as an issue instead of being answered by the docs.

Plan: locate the relevant section, confirm what the code actually does today (so the docs match behavior, not intent), tighten the wording, and scan neighboring pages for the same inconsistency — otherwise the fix just relocates the confusion.

One thing to confirm before writing: minimal correction scoped to this page, or also cross-link from related guides if the same gap shows up there?

(claiming via @LeonxLJX)
